src/main/java/com/esri/samples/graphics_update/position_sumulator/MessageGenerator.java

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright 2021 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.esri.samples.graphics_update.position_sumulator;
+
+import com.esri.arcgisruntime.geometry.Point;
+import java.io.*;
+import java.util.*;
+
+/**
+ * A class for generating simulated vehicle update messages
+ */
+public class MessageGenerator {
+
+    private HashMap<Integer, ArrayList<Point>> routes = new HashMap<>();
+    private ArrayList<Vehicle> vehicles = new ArrayList();
+    private UpdateMessageEventRunner updateMessageEventRunner = null;
+    private Timer timer;
+
+    /**
+     * Constructor for a new vehicle message generator.
+     * @param totalVehicles the number of vehicles messages will be generated for
+     */
+    public MessageGenerator(int totalVehicles) {
+        Random random = new Random();
+
+        // read in the route files
+        ReadRouteFiles();
+
+        // create vehicles
+        for (int vehicleID=1; vehicleID<=totalVehicles; vehicleID++ ) {
```

Comment:
readability enhancement
```suggestion
        for (int vehicleID = 1; vehicleID <= totalVehicles; vehicleID++ ) {
```